thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/AnomaliesResource.java

```diff
@@ -464,6 +467,69 @@ public void updateAnomalyMergedResultFeedback(@PathParam("mergedAnomalyId") long
     }
   }
 
+  /**
+   * Create a user-reported anomaly
+   *
+   * @param anomalyFunctionId anomaly function id (must exist)
+   * @param startTime start time utc (in millis)
+   * @param endTime end time utc (in millis)
+   * @param feedbackType anomaly feedback type
+   * @param comment anomaly feedback comment (optional)
+   * @param dimensionsJson dimension map json string (optional)
+   * @throws IllegalArgumentException if the anomaly function id cannot be found
+   * @throws IllegalArgumentException if the anomaly cannot be stored
+   */
+  @POST
+  @Path(value = "/reportAnomaly/{anomalyFunctionId}")
+  public void createUserAnomaly(@PathParam("anomalyFunctionId") long anomalyFunctionId,
+      @QueryParam("startTime") Long startTime,
+      @QueryParam("endTime") Long endTime,
+      @QueryParam("feedbackType") AnomalyFeedbackType feedbackType,
+      @QueryParam("comment") String comment,
+      @QueryParam("dimensionsJson") String dimensionsJson) {
+
+    AnomalyFunctionDTO anomalyFunction = anomalyFunctionDAO.findById(anomalyFunctionId);
+    if (anomalyFunction == null) {
+      throw new IllegalArgumentException(String.format("Could not resolve anomaly function id %d", anomalyFunctionId));
+    }
+
+    if (startTime == null) {
+      throw new IllegalArgumentException("Must provide startTime");
+    }
+
+    if (endTime == null) {
+      throw new IllegalArgumentException("Must provide endTime");
+    }
+
+    if (feedbackType == null) {
+      throw new IllegalArgumentException("Must provide feedbackType");
+    }
+
+    MergedAnomalyResultDTO anomaly = new MergedAnomalyResultDTO();
+    anomaly.setFunction(anomalyFunction);
+    anomaly.setStartTime(startTime);
+    anomaly.setEndTime(endTime);
+    anomaly.setDimensions(new DimensionMap(dimensionsJson != null ? dimensionsJson : "{}"));
+    anomaly.setAnomalyResultSource(AnomalyResultSource.USER_LABELED_ANOMALY);
+    anomaly.setMetric(anomalyFunction.getTopicMetric());
+    anomaly.setCollection(anomalyFunction.getCollection());
+    anomaly.setProperties(Collections.<String, String>emptyMap());
+    anomaly.setAnomalyResults(Collections.<RawAnomalyResultDTO>emptyList());
+
+    if (mergedAnomalyResultDAO.save(anomaly) == null) {
+      throw new IllegalArgumentException(String.format("Could not store user reported anomaly: '%s'", anomaly));
+    }
+
+    // TODO fix feedback not being saved on create by DAO
+    AnomalyFeedbackDTO feedback = new AnomalyFeedbackDTO();
+    feedback.setFeedbackType(feedbackType);
+    feedback.setComment(comment);
+
+    anomaly.setFeedback(feedback);
```

Comment:
Hi Alex, what's the "notified" flag for "user-report"? Here are two use cases:
1. In evaluation module, it relies on "isNotified=false" to make classify user-report anomalies
2. In search page (alert page), when showing anomalies is depending on "notified = true", but user report anomalies should also be shown on the page. You probably need to edit the logic of showing searched anomalies. (including user report, replayed anomalies)

Reply:
@cecilynie yes this is another thing to change. it's for a separate PR.